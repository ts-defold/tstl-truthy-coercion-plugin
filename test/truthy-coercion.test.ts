import * as util from "./util";

test("truthy strings", () => {
	util.testFunction`
		const foo = "";
		if (foo) {
			return true;
		}
		else return false;
	`.expectToMatchJsResult();
});

test("falsy string logical and", () => {
	util.testFunction`
		return "" || "dog";
	`.expectToMatchJsResult();
});

test("truthy numbers", () => {
	util.testFunction`
		const foo = 0;
		const bar = NaN;
		const results = [];

		if (foo) { results.push(true); } else { results.push(false); }
		if (bar) { results.push(true); } else { results.push(false); }

		return results;
	`.expectToMatchJsResult();
});

test("falsy number logical and", () => {
	util.testFunction`
		return 0 || 100;
	`.expectToMatchJsResult();
});

test("falsy", () => {
	util.testFunction`
		const results = [];

		if (false) results.push(1);
		if (null) results.push(2);
		if (undefined) results.push(3);
		if (0) results.push(4);
		if (-0) results.push(5);
		if (NaN) results.push(6);
		if ("") results.push(7);

		return results;
	`.expectToMatchJsResult();
});

test("truthy", () => {
	util.testFunction`
		const results = [];

		if (true) results.push(1);
		if ({}) results.push(2);
		if ([]) results.push(3);
		if (42) results.push(4);
		if ("0") results.push(5);
		if ("false") results.push(6);
		if (-42) results.push(7);
		if (3.14) results.push(8);
		if (-3.14) results.push(9);
		if (Infinity) results.push(10);
		if (-Infinity) results.push(11);

		return results;
	`.expectToMatchJsResult();
});
