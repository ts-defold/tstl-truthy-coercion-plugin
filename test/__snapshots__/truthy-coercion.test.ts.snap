// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`falsy number logical and 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    return 0 or 100
end
return ____exports"
`;

exports[`falsy string logical and 1`] = `
"local ____exports = {}
function ____exports.__main(self)
    return \\"\\" or \\"dog\\"
end
return ____exports"
`;
